src/test/java/io/gingersnapproject/service/MembershipListenerTest.java

```diff
@@ -0,0 +1,86 @@
+package io.gingersnapproject.service;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.clearInvocations;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.net.InetAddress;
+import java.net.URI;
+import java.util.HashSet;
+import java.util.List;
+
+import javax.enterprise.inject.Instance;
+
+import io.gingersnapproject.cdc.configuration.Cache;
+import io.gingersnapproject.cdc.configuration.Configuration;
+import io.gingersnapproject.util.Utils;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
Did you mean to have multiple test methods?

Reply:
Only this one should cover everything. I tried an additional one calling `onStartup`, but it has to wait until the task is scheduled.

Reply:
Okay, was just wondering because this annotation is worthless right now otherwise.